File: Solutions/0442- Find All Duplicates in an Array/Solution.java
class Solution {
    public List<Integer> findDuplicates(int[] nums) {
        
        for (int i = 0; i < nums.length; ++i) {
            while (nums[i] != nums[nums[i] - 1]) {
                var temp = nums[nums[i] - 1];
                nums[nums[i] - 1] = nums[i];
                nums[i] = temp;
            }
        }
        List<Integer> answer = new ArrayList<>();

        for (int i = 0; i < nums.length; i++) {
            if (nums[i] != i +1) {
                answer.add(nums[i]);
            }
        }

        return answer;
    }
}

File: Solutions/0442- Find All Duplicates in an Array/Solution.py
class Solution:
    def findDuplicates(self, nums: List[int]) -> List[int]:
        answer = list()

        for i in range(0,len(nums)):
            while nums[i] != nums[nums[i] - 1]:
                temp = nums[nums[i] - 1]
                nums[nums[i] - 1] = nums[i]
                nums[i] = temp
        
        for i in range(0,len(nums)):
            if nums[i] != i + 1:
                answer.append(nums[i])
        
        return answer
        